dask_cuda/explicit_comms/dataframe/shuffle.py

```diff
@@ -23,9 +22,72 @@
 T = TypeVar("T")
 
 
+class Proxify(Protocol):
+    """Proxify type hint"""
+
+    def __call__(self, obj: T) -> T:
+        ...
+
+
+def get_proxify(worker: Worker) -> Proxify:
+    """Get function to proxify objects"""
+    from dask_cuda.proxify_host_file import ProxifyHostFile
+
+    if isinstance(worker.data, ProxifyHostFile):
+        data = worker.data
+        return lambda x: data.manager.proxify(x)[0]
+    return lambda x: x  # no-op
+
+
+def get_no_comm_postprocess(
+    stage: Dict[str, Any], num_rounds: int, batchsize: int
+) -> Callable[[DataFrame], DataFrame]:
+    """Get function for post-processing partitions not communicated
+
+    In cuDF, the `group_split_dispatch` uses `scatter_by_map` to create
+    the partitions, which is implemented by splitting a single base dataframe
+    into multiple partitions. This means that memory are not freed until
+    ALL partitions are deleted.
+
+    In order to free memory ASAP, we can deep copy partitions NOT being
+    communicated. We do this when `num_rounds != batchsize`.
+
+    Parameters
+    ----------
+    stage
+        The staged input dataframes.
+    num_rounds: int
+        Number of rounds of dataframe partitioning and all-to-all communication.
+    batchsize: int
+        Number of partitions each worker will handle in each round.
+
+    Returns
+    -------
+    Function to be called on partitions not communicated.
+
+    """
+    if num_rounds == batchsize:
+        return lambda x: x
+
+    # Check that we are shuffling a cudf dataframe
+    try:
+        import cudf
+    except ImportError:
+        return lambda x: x
+    if not stage or not isinstance(next(iter(stage.values())), cudf.DataFrame):
+        return lambda x: x
+
+    # Deep coping a cuDF dataframe doesn't deep copy its index hence
```

Comment:
```suggestion
    # Deep copying a cuDF dataframe doesn't deep copy its index hence
```

Q: is this a cudf bug?